spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/SparkTemplate.scala

```diff
@@ -1,25 +1,56 @@
 package com.datastax.spark.connector.embedded
 
-import org.apache.spark.{SparkEnv, SparkConf, SparkContext}
+import akka.actor.ActorSystem
+import org.apache.spark.{SparkConf, SparkContext, SparkEnv}
 
 trait SparkTemplate {
-  val conf = SparkTemplate.conf
-  val sc = SparkTemplate.sc
-}
-
-object SparkTemplate {
+  /** Obtains the active [[SparkContext]] object. */
+  def sc: SparkContext = SparkTemplate.sc
 
-  val conf = new SparkConf(true)
-    .set("spark.cassandra.connection.host", EmbeddedCassandra.cassandraHost.getHostAddress)
-    .set("spark.cleaner.ttl", "3600")
-    .set("spark.metrics.conf", this.getClass.getResource("/metrics.properties").getPath)
-    .setMaster(sys.env.getOrElse("IT_TEST_SPARK_MASTER", "local[*]"))
-    .setAppName(getClass.getSimpleName)
+  /** Obtains the [[ActorSystem]] associated with the active [[SparkEnv]]. */
+  def actorSystem: ActorSystem = SparkTemplate.actorSystem
 
-  System.err.println("The Spark configuration is as follows:\n" + conf.toDebugString)
+  /** Ensures that the currently running [[SparkContext]] uses the provided configuration. If the
+    * configurations are different or force is `true` Spark context is stopped and started again with
+    * the given configuration. */
+  def useSparkConf(conf: SparkConf = SparkTemplate.defaultConf, force: Boolean = false): SparkContext =
+    SparkTemplate.useSparkConf(conf, force)
+}
 
-  val sc = new SparkContext(conf)
+object SparkTemplate {
 
-  lazy val actorSystem = SparkEnv.get.actorSystem
+  /** Default configuration for [[SparkContext]]. */
+  val defaultConf = new SparkConf(true)
+                    .set("spark.cassandra.connection.host", EmbeddedCassandra.cassandraHost.getHostAddress)
+                    .set("spark.cleaner.ttl", "3600")
+                    .setMaster(sys.env.getOrElse("IT_TEST_SPARK_MASTER", "local[*]"))
+                    .setAppName("Test")
+
+  /** Ensures that the currently running [[SparkContext]] uses the provided configuration. If the
+    * configurations are different or force is `true` Spark context is stopped and started again with
+    * the given configuration. */
+  def useSparkConf(conf: SparkConf = SparkTemplate.defaultConf, force: Boolean = false): SparkContext = {
```

Comment:
Conf shouldn't have a default value here. If you insist on not giving the default argument explicitly, a special `useDefaultSparkConf` would be better.